### ISSUE TITLE
Even minimized and with no data loaded CPU is constantly being used on windows

### ISSUE THREAD

**Issue by wchristian** [desktop/desktop#1641]
### Description

As the title says. It is minimized, has no data loaded, yet constantly uses CPU. To be clear on no data loaded, this was directly after install, with only the login being provided.

![image](https://cloud.githubusercontent.com/assets/175467/26239002/ae503026-3c7b-11e7-9d9b-75dad24f17fe.png)

![image](https://cloud.githubusercontent.com/assets/175467/26239304/c0268ca4-3c7c-11e7-80b7-9be730d1c1a7.png)

Note that the usage must be multiplied by 4 to get the load on a single CPU core, as i have 4 cores.

### Version

**GitHub Desktop version:** 0.5.5

**OS version:** Microsoft Windows [Version 10.0.15063]

### Steps to Reproduce

1. Install Github Desktop
1. Start Github Desktop
1. Log in
1. Minimize
1. Observe CPU usage with [Technet Process Explorer](https://technet.microsoft.com/en-us/sysinternals/processexplorer.aspx)

**Expected behaviour:** All of the Github Desktop processes should read a usage of 0 for at least a minute between usage spikes

**Actual behaviour:** Github Desktop is *constantly* using CPU

**Reproduces how often:** 100%

**Comment by LeoNatan** [comment]
Why does a git UI application need to have 4 processes running, each taking 25-55MB of RAM at least?

Ah, the wonders of Electron "development". Good stuff, "native" "developers".

**Comment by shiftkey** [comment]
@LeoNatan the reasons for this are explained in the [Multi-process Architecture](https://www.chromium.org/developers/design-documents/multi-process-architecture) documentation on the Chromium project - the "browser" and "renderer" separations here flow through to the various Electron APIs.

**Comment by shiftkey** [comment]
Yep, it'd be nice to dig into this and see what things we can do in-app, versus working more closely with the Electron core team if it's outside our control. 

cc @desktop/electron for feels

**Comment by LeoNatan** [comment]
@shiftkey Oh, I understand why Chromium/Chrome, aka "browser", needs a multi-process architecture. But what genius figured out it was also needed for a git UI application? Or a collaboration service app, or a text editor, for that matter? Lazy companies, that's who.

**Comment by joshaber** [comment]
@LeoNatan We welcome constructive feedback but the tone of your comments violates our [Code of Conduct](https://github.com/desktop/desktop/blob/2280d0707c40a18cb992a22ffa596e997840e6dd/CODE_OF_CONDUCT.md). This is your warning. If you continue you will be banned.

**Comment by lee-dohm** [comment]
@LeoNatan your comment was deleted as a violation of the [Desktop Code of Conduct](https://github.com/desktop/desktop/blob/master/CODE_OF_CONDUCT.md) as is insulting or derogatory. You may consider this an official warning.

**Comment by wchristian** [comment]
@shiftkey I suspect there are at least some things you could do. I've observed VS Code, while not being a good citizen either, being a better citizen.

**Comment by Mr-BrainSpace** [comment]
I had this same issue when I first installed, even after closing the desktop application on Windows 7 I had processes running up the CPU. At the same time a process called Update.exe was being suspicious. I found reopening Github Desktop, and going File > Close -- vs the usual red [X] -- got rid of both problems. Could be a coincidence.

![image](https://user-images.githubusercontent.com/30075973/28076139-b4b5dcfc-662b-11e7-894d-86ef454b93e6.png)

**Comment by wchristian** [comment]
Still happens.

**Comment by shiftkey** [comment]
@wchristian I had an initial look at this a few months ago - it basically needs you to have a debug version of Electron and then know how to profile external processes in Visual Studio, but I got stuck at having to also provide a debug version of Node (where there was some interesting activity but lacked the symbols to correlate).

What I did find when the app was minimized and in the background was a bunch of message pump activity, which was to be expected. So yeah, more research needed. I'll see if I can write up the process I went through to get to that point - maybe someone else is keen to get involved here.

**Comment by wchristian** [comment]
@shiftkey Just documenting that sounds like an excellent step! I don't know if what i'm seeing is easily reproducable on other machines, so if you give me a build i can provide profiling data with i'd happily do that.

**Comment by hulk0815** [comment]
> ### Description
> As the title says. It is minimized, has no data loaded, yet constantly uses CPU. To be clear on no data loaded, this was directly after install, with only the login being provided.
> 
> ![image](https://cloud.githubusercontent.com/assets/175467/26239002/ae503026-3c7b-11e7-9d9b-75dad24f17fe.png)
> 
> ![image](https://cloud.githubusercontent.com/assets/175467/26239304/c0268ca4-3c7c-11e7-80b7-9be730d1c1a7.png)
> 
> Note that the usage must be multiplied by 4 to get the load on a single CPU core, as i have 4 cores. Also note that this kind of usage is significant as it prevents the CPU from settling in a low power state.
> 
> ### Version
> **GitHub Desktop version:** 0.5.5
> 
> **OS version:** Microsoft Windows [Version 10.0.15063]
> 
> ### Steps to Reproduce
> 1. Install Github Desktop
> 2. Start Github Desktop
> 3. Log in
> 4. Minimize
> 5. Observe CPU usage with [Technet Process Explorer](https://technet.microsoft.com/en-us/sysinternals/processexplorer.aspx)
> 
> **Expected behaviour:** All of the Github Desktop processes should read a usage of 0 for at least a minute between usage spikes
> 
> **Actual behaviour:** Github Desktop is _constantly_ using CPU
> 
> **Reproduces how often:** 100%
